src/main/java/com/bookstore/service/ShoppingCartServiceImpl.java

```diff
@@ -0,0 +1,107 @@
+package com.bookstore.service;
+
+import com.bookstore.dto.CartItemRequestDto;
+import com.bookstore.dto.CartItemResponseDto;
+import com.bookstore.dto.CartItemUpdateRequestDto;
+import com.bookstore.dto.ShoppingCartResponseDto;
+import com.bookstore.exception.EntityNotFoundException;
+import com.bookstore.mapper.CartItemMapper;
+import com.bookstore.mapper.ShoppingCartMapper;
+import com.bookstore.model.Book;
+import com.bookstore.model.CartItem;
+import com.bookstore.model.ShoppingCart;
+import com.bookstore.model.User;
+import com.bookstore.repository.BookRepository;
+import com.bookstore.repository.CartItemRepository;
+import com.bookstore.repository.ShoppingCartRepository;
+import com.bookstore.repository.UserRepository;
+import lombok.RequiredArgsConstructor;
+import org.springframework.security.core.context.SecurityContextHolder;
+import org.springframework.security.core.userdetails.UsernameNotFoundException;
+import org.springframework.stereotype.Service;
+import org.springframework.transaction.annotation.Transactional;
+
+@Service
+@RequiredArgsConstructor
+public class ShoppingCartServiceImpl implements ShoppingCartService {
+
+    private final ShoppingCartRepository shoppingCartRepository;
+    private final CartItemRepository cartItemRepository;
+    private final BookRepository bookRepository;
+    private final UserRepository userRepository;
+    private final ShoppingCartMapper shoppingCartMapper;
+    private final CartItemMapper cartItemMapper;
+
+    @Override
+    @Transactional(readOnly = true)
+    public ShoppingCartResponseDto getCartForCurrentUser() {
+        User user = getCurrentUser();
+        ShoppingCart cart = shoppingCartRepository.findByUser(user)
+                .orElseGet(() -> createCartForUser(user));
```

Comment:
see comment for createCartForUser
```suggestion
                .orElseThrow()
```